sunpy/net/dataretriever/sources/secchi.py

```diff
@@ -0,0 +1,72 @@
+import socket
+from datetime import datetime
+
+from sunpy.extern.parse import parse
+from sunpy.net.dataretriever import GenericClient, QueryResponse
+from sunpy.util.scraper import Scraper
+from sunpy.time import TimeRange, parse_time
+
+
+__all__ = ['SECCHIlient']
+
+
+class SECCHIlient(GenericClient):
+    """
+    Provides access to the STEREO/SECCHI data.
+
+    Uses this `archive <https://stereo-ssc.nascom.nasa.gov/pub/beacon/ahead/secchi/img>`.
+    """
+
+    patterns = {
+        'euvi': '{}/img/euvi/{year:4d}{month:2d}{day:2d}/{year:4d}{month:2d}{day:2d}_{}',
+        'cor2': '{}/img/cor2/{year:4d}{month:2d}{day:2d}/{year:4d}{month:2d}{day:2d}_{}',
+        'hi_1': '{}/img/hi_1/{year:4d}{month:2d}{day:2d}/{year:4d}{month:2d}{day:2d}_{}',
+        'hi_2': '{}/img/hi_2/{year:4d}{month:2d}{day:2d}/{year:4d}{month:2d}{day:2d}_{}'
+    }
+    baseurl = 'https://stereo-ssc.nascom.nasa.gov/pub/beacon/ahead/secchi/img'
+
+    def get_filenames(self, timerange):
+        telescopes = ['euvi', 'cor2', 'hi_1', 'hi_2']
+        tele_url = {
+            'euvi': 'n7euA',
+            'cor2': 'd7c2A',
+            'hi_1': 's7h1A',
+            'hi_2': 's7h2A'
+        }
+
+        filenames = {}
+        for telescope in telescopes:
```

Comment:
similarly, you would probably want an `attrs` here to choose which telescope you want